bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1135 @@
+        try {
+            Completable<@Nullable Stream> streamPromise = new Completable<>();
+            ContentStreamListenerAdapter contentStreamListener = new ContentStreamListenerAdapter();
+            session.newStream(headers, streamPromise, contentStreamListener);
+            // wait for stream to be opened
+            Objects.requireNonNull(streamPromise.get(TIMEOUT_SECONDS, TimeUnit.SECONDS));
+            // wait for HTTP response contents
+            String contentJson = contentStreamListener.awaitResult();
+            String contentType = contentStreamListener.getContentType();
+            logger.trace("HTTP/2 200 OK (Content-Type: {}) << {}", contentType, contentJson);
+            if (!MediaType.APPLICATION_JSON.equals(contentType)) {
+                throw new ApiException("Unexpected Content-Type: " + contentType);
+            }
+            try {
+                Resources resources = Objects.requireNonNull(jsonParser.fromJson(contentJson, Resources.class));
+                if (logger.isDebugEnabled()) {
+                    resources.getErrors().forEach(error -> logger.debug("Resources error:{}", error));
+                }
+                return resources;
+            } catch (JsonParseException e) {
+                throw new ApiException("Parsing error", e);
+            }
+        } catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof HttpUnauthorizedException) {
+                throw (HttpUnauthorizedException) cause;
+            }
+            throw new ApiException("Error sending request", e);
+        } catch (TimeoutException e) {
+            throw new ApiException("Error sending request", e);
+        } finally {
+            throttleDone();
+        }
+    }
+
+    /**
+     * Build a full path to a server end point, based on a Reference class instance. If the reference contains only
+     * a resource type, the method returns the end point url to get all resources of the given resource type. Whereas if
+     * it also contains an id, the method returns the end point url to get the specific single resource with that type
+     * and id.
+     *
+     * @param reference a Reference class instance.
+     * @return the complete end point url.
+     */
+    private String getUrl(ResourceReference reference) {
+        String url = baseUrl + reference.getType().name().toLowerCase();
+        String id = reference.getId();
+        return Objects.isNull(id) || id.isEmpty() ? url : url + "/" + id;
+    }
+
+    /**
+     * Restart the session.
+     *
+     * @param active boolean that selects whether to restart in active or passive mode.
+     */
+    private void internalRestart(boolean active) {
+        internalRestartScheduled = false;
+        try {
+            openPassive();
+            if (active) {
+                openActive();
+            }
+        } catch (ApiException e) {
+            logger.warn("scheduledReconnectTask() failed ", e);
```

Comment:
Can this warning be made more user-oriented?

Reply:
And without full stack trace - see https://github.com/openhab/openhab-addons/pull/13570#discussion_r1244331935

Reply:
Ok. Done.